doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+right angle even in case of upside-down face arrangment (if we put the points in
+the right order, of course).
+
+Back to the `getEyeEllipse()` function, it returns a constructed `Contour`
+to use the RVO.
+
+After the bottom side of an eye is approximated, we just push the eyebrow points
+to the contour in the right order - and the eye is finished!
+
+#### Mouth contour getting {#gapi_fb_ld_mth}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp ld_pp_mth
+
+We have four points of a mouth given: two of them in corners and two others in
+the middle of upper and lower lip. So the mouth can be approximated by two
+patched half-ellipses:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_pp_mth
+
+At this point Y half-axes are defined without any assumptions, but they are
+different for the upper lip and the lower one.
+
+#### Forehead contour getting {#gapi_fb_ld_fhd}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp ld_pp_fhd
+
+This snippet describes two actions: the approximation of a forehead by
+half-ellipse and pushing the jaw contour to the array in the appropriate order.
+
+The function to approximate a forehead:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_pp_fhd
+
+As we have only jaw points, we are trying to get a half-ellipse based on only
+three points of a jaw: the leftmost, the rightmost and the lowest. Obviously,
+the jaw width is equal to the forehead width and can be counted by the left and
+right points. What about the Y axis, we haven't points to get it directly;
+but, according to an assumption, the forehead height is about 2/3 of the jaw
+height, which can be received from the face center (the middle between the left
+and right points) and the lowest jaw point.
+
+We have all the counters needed! Let's draw masks now.
+
+## Masks drawing {#gapi_fb_masks_drw}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp msk_ppline
+
+Honestly saying, there is no need to get the most accurate contours in the
+world: we can always blur borders by gaussian filter to occupy a bit more area.
+Another idea is to get three masks which are for bilateral-filtered image, for
+sharped image and for the untouched one and have no intersections with each
+other; this approach allows just to summarize images with the masks applied and
+get the output without any other operations:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp msk_appl
+
+Thus, the steps to get the masks are:
+* fill the contours that should be sharpened;
+* blur that to get the "sharp" mask;
+* fill all the face contours fully;
+* blur that;
+* subtract areas which intersect with the "sharp" mask --- and get the
+"bilateral" mask;
+* now get the mask of all the area that shouldn't be touched by filters
```

Comment:
Can split this point to several smaller steps. Remove `now`

Reply:
rewritten